libraries/AP_HAL_Linux/RCOutput_PCA9685.cpp

```diff
@@ -225,16 +225,11 @@ void RCOutput_PCA9685::push()
         *d++ = length >> 8;
     }
 
-    if (!_dev || !_dev->get_semaphore()->take_nonblocking()) {
```

Comment:
This looks kind of dangerous.

Firstly, there's a whole bunch of other places that are doing nullptr checks on _dev - what's the reasoning that it's not required here?

Secondly, why is the bus semaphore not required in this case?  

Not taking the semaphore means if there's anything else on the bus you may corrupt everything.

ChibiOS wouldn't actually allow you to do this at all - if the thread doesn't own the bus semaphore you can't `transfer(...)` - we should add an owner to the bus in `AP_HAL_Linux` (I recently added it to `AP_HAL_SITL`).  Thoughts @lucasdemarchi ?

Reply:
In Linux the transfer() call will basically block until the first thread returns. Remember that could something even outside of ardupilot playing with the I2C bus. Anyway, in Linux we also implement the mutex and _dev->get_semaphore() returns the semaphore for that bus (not the device). See  libraries/AP_HAL_Linux/I2CDevice.cpp

If that check you are removing fails, then there was an error in how it was initialized. See in the constructor that we std::move() whatever was passed as argument.

The reason for the additional `!dev` sprinkled is that this is RCOuptut, not a optional sensor.